src/devices/Ak8963/Ak8963.cs

```diff
@@ -0,0 +1,262 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.IO;
+using System.Numerics;
+using System.Threading;
+
+namespace Iot.Device.Ak8963
+{
+    /// <summary>
+    /// AK8963 class implementing a magnetometer
+    /// </summary>
+    public class Ak8963 : IDisposable
+    {
+        private I2cDevice _i2cDevice;
+        private MeasurementMode _measurementMode;
+        private OutputBitMode _outputBitMode;
+        private bool _selfTest = false;
+        private Ak8963I2cBase _ak8963Interface;
+        private bool _autoDispose;
+
+        public const byte DefaultI2cAddress = 0x0C;
+
+        public Ak8963(I2cDevice i2CDevice) : this(i2CDevice, new Ak8963I2c())
```

Comment:
If you do make the other constructor internal as I suggest, then you should change the current conditional property `bool autoDispose = true` and make it a regular parameter. Then you would always autodispose when controlling through I2c, and never dispose when you use from Mpu.

Reply:
See my previous comment, because they are in separate name space, can't make the constructor internal